sencha-workspace/SlateTasksStudent/app/controller/Dashboard.js

```diff
@@ -87,13 +90,22 @@ Ext.define('SlateTasksStudent.controller.Dashboard', {
         'combo#section-selector': {
             select: 'onSectionSelectorSelect',
             boxready: 'onSectionSelectorBoxReady'
+        },
+        'combo#student-selector': {
+            select: 'onStudentSelectorSelect',
+            boxready: 'onStudentSelectorBoxReady'
         }
     },
 
 
     // controller templates method overrides
     onLaunch: function () {
         this.getDashboard().render('slateapp-viewport');
+
+        // hide student selector from students
+        if (SiteEnvironment.user && SiteEnvironment.user.AccountLevel == "User") {
```

Comment:
this needs to be prefixed with `window.SiteEnvironment &&` to avoid crashing if the global isn't set. `window.` doesn't need to be used after that if it is defined. And don't use double quotes for strings when single quotes are used everywhere else.

Reply:
Also, it's better to configure the component as `hidden: true` initially and then `setHidden(false)` if it should be shown, that way there is never a flash of it appearing.

Reply:
@themightychris the window part makes sense.

I set the hidden logic how I did because if you're developing locally, window.SiteEnvironment.user doesn't exist meaning the dropdown will never actually be visible. Obviously this won't be a problem in production but it means you have to manually set hidden = false just to test this feature. Having a step like that in place to me seem like it could be prone to error or at least a dev accidentally committing the hidden flag accidentally.